### PR TITLE
Add start of game logic / design

### PR DESCRIPTION
**Pull request by maggiemay-afk**
Few small changes for start of game logic: 

- Transparent overlay with game instructions and start game button
This should fix the bug / issue with the browser not liking the auto play music 

- Disable YouTube Player buttons when gameStarted state is false
This required updating our button wrapper to allow for the "disabled" field and new global state, gameStarted

- Move 'isPlaying' state in YoutubePlayer component up to global state, in state.ts
This is so that the new StartGame component can set 'isPlaying' state to true. This will "auto" start the music after a user clicks on the 'start' button in the StartGame component. 

Let me know if you have any design feedback or feedback on the instruction wording!

### CODE REVIEW

**Comment by maggiemay-afk** [comment]
@hermanator608 Expanding this PR slightly, wanted to include some more style changes and game round logic

- Remove the window alert to ask a user if the area selected is their final choice
- Change it to MUI alert with 'yes' and 'no' buttons
- on click of 'Yes' will evaluate user choice with the same logic as before
- on click of 'No' will remove alert and do nothing else (not ending the round)
- Added the correct zone answer (i.e., zone name) and more styling to the display after a user makes their final choice
- Other, related styling

File: src/App.tsx [hunk 1]
```diff
@@ -15,14 +17,21 @@ import ImageMapper, { ImageMapperProps, MapAreas } from 'react-img-mapper';
...
+import ButtonGroup from '@mui/material/ButtonGroup';
 
 
 function App() {
   const [currentQuestion, setCurrentQuestion] = useRecoilState(currentQuestionState);
   const [videoShown, setVideoShown] = useRecoilState(videoShownState);
   const [world, setWorld] = useState<Stack<MapDetails>>(new Stack<MapDetails>(5, AllWorlds));
   const [hoverArea, setHoverArea] = useState<String>();
-  const [result, setResult] = useState<Boolean>();
+  const [result, setResult] = useState<String>();
```
Comment (hermanator608) [inline src/App.tsx]: We could update this to a boolean instead of a string, might be cleaner?

File: src/App.tsx [hunk 2]
```diff
@@ -15,14 +17,21 @@ import ImageMapper, { ImageMapperProps, MapAreas } from 'react-img-mapper';
...
+  const [answer, setAnswer] = useState<String>("");
+  const [gameStarted, ] = useRecoilState(gameStartedState);
+  const [showAlert, setShowAlert] = useState<Boolean>(false);
+  const [choice, setChoice] = useState<String>("");
```
Comment (hermanator608) [inline src/App.tsx]: Maybe do `String | undefined` to make it clearer that there isnt a selection yet

File: src/App.tsx [hunk 3]
```diff
@@ -15,14 +17,21 @@ import ImageMapper, { ImageMapperProps, MapAreas } from 'react-img-mapper';
...
 
 
 function App() {
   const [currentQuestion, setCurrentQuestion] = useRecoilState(currentQuestionState);
   const [videoShown, setVideoShown] = useRecoilState(videoShownState);
   const [world, setWorld] = useState<Stack<MapDetails>>(new Stack<MapDetails>(5, AllWorlds));
   const [hoverArea, setHoverArea] = useState<String>();
-  const [result, setResult] = useState<Boolean>();
+  const [result, setResult] = useState<String>();
+  const [answer, setAnswer] = useState<String>("");
```
Comment (hermanator608) [inline src/App.tsx]: Maybe do String | undefined to make it clearer that there isnt an answer yet

File: src/App.tsx [hunk 4]
```diff
@@ -92,28 +114,51 @@ function App() {
...
+      >
+      {result}: {answer}
+      </Typography>
+
+      :  <div className="photo">
             <ImageMapper {...imageMapperProps} />
+            {!gameStarted
+              ?<StartGame />
+              :<></> 
+            }
```
Comment (hermanator608) [inline src/App.tsx]: Can update this to be:
```
{!gameStarted && <StartGame />}
```

File: src/App.tsx [hunk 5]
```diff
@@ -92,28 +114,51 @@ function App() {
...
             <IconButton 
               className="back-button"
               size="medium" 
               sx={{position:"absolute", backgroundColor: "white"}}
               disabled={!!videoShown || world.peek() === AllWorlds} 
-              onClick={() => { removeMap(); }} 
+              onClick={() => { removeMap(); }}
```
Comment (hermanator608) [inline src/App.tsx]: nitpick: Can be cleaned up to
```
onClick={removeMap}
```

File: src/App.tsx [hunk 6]
```diff
@@ -92,28 +114,51 @@ function App() {
...
-
+            {!showAlert
+            ? <></>
+            : <Alert 
+                className="alert-parent"
+                severity="warning"
+                action={
+                  <ButtonGroup>
+                    <Button color="inherit" size="small" onClick={() => {evaluateChoice();}}>YES</Button>
+                    <Button color="inherit" size="small" onClick={() => {removeAlert();}}>NO</Button>
```
Comment (hermanator608) [inline src/App.tsx]: Should be able to clean these up similar to above comment

File: src/App.tsx [hunk 7]
```diff
@@ -15,14 +17,21 @@ import ImageMapper, { ImageMapperProps, MapAreas } from 'react-img-mapper';
...
 
 function App() {
   const [currentQuestion, setCurrentQuestion] = useRecoilState(currentQuestionState);
   const [videoShown, setVideoShown] = useRecoilState(videoShownState);
   const [world, setWorld] = useState<Stack<MapDetails>>(new Stack<MapDetails>(5, AllWorlds));
   const [hoverArea, setHoverArea] = useState<String>();
-  const [result, setResult] = useState<Boolean>();
+  const [result, setResult] = useState<String>();
+  const [answer, setAnswer] = useState<String>("");
+  const [gameStarted, ] = useRecoilState(gameStartedState);
```
Comment (hermanator608) [inline src/App.tsx]: Can update this to use https://recoiljs.org/docs/api-reference/core/useRecoilValue:
```
const gameStarted = useRecoilValue(gameStartedState);
```

File: src/components/StartGame.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import { useRecoilState } from "recoil";
+import Box from '@mui/material/Box';
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+import { gameStartedState, isPlayingState } from "../state";
+
+
+function StartGame(){
+  const [, setGameStarted] = useRecoilState(gameStartedState);
+  const [, setIsPlaying] = useRecoilState(isPlayingState);
```
Comment (hermanator608) [inline src/components/StartGame.tsx]: Can update these to use https://recoiljs.org/docs/api-reference/core/useSetRecoilState
e.g.
```
  const setGameStarted = useSetRecoilState(gameStartedState);
```

File: src/components/YoutubePlayer.tsx [hunk 9]
```diff
@@ -105,16 +107,18 @@ export const YoutubePlayer: React.FC<{soundtrackIndex?: number}> = ({
   // Global State
   const [videoShown] = useRecoilState(videoShownState);
   const [currentQuestion] = useRecoilState(currentQuestionState);
+  const [isPlaying, setIsPlaying] = useRecoilState(isPlayingState);
+  const [gameStarted, ] = useRecoilState(gameStartedState);
```
Comment (hermanator608) [inline src/components/YoutubePlayer.tsx]: https://recoiljs.org/docs/api-reference/core/useRecoilValue

**Review by hermanator608 — COMMENTED** [review]
Few small comments! Looking good 👍 

Few UI suggestions that would be nice to have but can be done in a different PR:
* Fade transition animation
* Along with fade reduce the screen jumping when selecting your final choice and when youtube video stops and goes back to map

**Comment by maggiemay-afk** [comment]
Okay - updates made based on your comments!

**Review by hermanator608 — APPROVED** [review]

**Pull request merged**